### PR TITLE
Carlos anniversary celebrated

### PR DESCRIPTION
**Pull request by cgomezhub**
# Description
This part of a new Report Page: Create Company Summary Dashboard for Admins requested by Jae.
Implements # (WBS)
Anniversary Celebrared Component

## Related PRS (if any):

This frontend PR is related to the [#1128](https://github.com/OneCommunityGlobal/HGNRest/pull/1128) backend PR.

Screenshot 2024-10-10 091351

Screenshot 2024-10-10 091516

Screenshot 2024-10-10 091617

…

## Main features  explained:
- Visualization of volunteer anniversaries within the current week.
- Comparison of volunteer anniversaries between the current and previous week.
- Clicking on the volunteer email icon allows you to send a personalized congratulatory email using previously configured Gmail OAuth settings. (For more information on Gmail setup, please watch the video below.)

…

## How to test:

This frontend PR is related to the [#1128](https://github.com/OneCommunityGlobal/HGNRest/pull/1128) backend PR.

1. Check out the current branch.
2. Do `npm install` and `...` to run this PR locally.
3. Clear site data/cache.
4. Log in  as admin user.
5. Navigate to Dashboard > Reports > Total Org Summary > Volunteer Engagement Trends.
6. Set up the  Gmail Api in your account  http://console.cloud.google.com and https://developers.google.com/  (please watch   the Gmail Api setup video belowuntil minut 7).
7.Set up the environment variables in the .env file of your backend project (see .env image).
8. Chech the anniversary list.
9. Select and click on the email icon.
10. Send a congratulatory email.
11. Check sent folder of your gmail acount.

### App features video:

https://www.loom.com/share/d4f75956df7b4ef6946dd17aa8bfa806?sid=5ccabce1-1fb8-4c11-a9f7-a5127a3ea793

### Gmail Api setup video (just watch until minut 7)

https://www.youtube.com/watch?v=-rcRf7yswfM&list=LL&index=8&t=304s

### Image of Environment variables

Screenshot 2024-10-10 093847

## Note:
Remember!. Here are  the absolutely necessary steps to successfully send a congratulatory email from your Google account:

- Set up the Gmail API in your account. (Removes unnecessary "Api") - Links can stay as provided. (Please watch the Gmail API setup video  until minute 7).
- Set up the environment variables in the .env file of your backend project. (See  image)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,6 +99,7 @@
     "@testing-library/jest-dom": "^5.11.3",
     "@testing-library/react": "^10.4.3",
     "@testing-library/user-event": "^12.0.14",
+    "@types/react-router-dom": "^5.3.3",
```
Comment (Ankuriboh) [inline package.json:102]: Adding a new `devDependency` should be noted in the PR, even though it's just a typing library to facilitate developing.

**Review by Ankuriboh — APPROVED** [review]
Can confirm that sending anniversary congratulation emails works on my end. A small problem I found is that when sending email while I didn't edit content in the editor, the system will send an empty email.

This feature can be further enhanced by adding scheduled emails sending (i.e. automated congrats email on n-year + their registered date), and probably some template email text for congratulating the user.

File: src/actions/totalOrgSummary.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ export const fetchTotalOrgSummaryReportError = error => ({
 export const getTotalOrgSummary = (startDate, endDate) => {
   const url = ENDPOINTS.TOTAL_ORG_SUMMARY(startDate, endDate);
   return async dispatch => {
-    dispatch(fetchTotalOrgSummaryReportBegin());
+    await dispatch(fetchTotalOrgSummaryReportBegin());
```
Comment (sheetalmangate) [inline src/actions/totalOrgSummary.js:34]: The return value of promise is not saved in any constant.  what is the use of using await operator?

**Review by zhiminV — DISMISSED** [review]
I have check the feature inLight and dark mode and it work as expected

https://github.com/user-attachments/assets/f6567020-b905-43d0-810b-d734bed85d4f

File: package.json (hunk 1 above)
Comment (cgomezhub) [reply package.json:102]: thanks.
that will be part of a next improvement of the feature

File: src/actions/totalOrgSummary.js (hunk 2 above)
Comment (cgomezhub) [reply src/actions/totalOrgSummary.js:34]: Actually, that code is not involved in this feature enhancement. It was a typo that will be fixed. Thanks.

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**